### ISSUE TITLE
IOS Xcode simulator: "You aren't under Cordova environment"

### ISSUE THREAD

**Issue by DavidFishman** [jsep/meteor-facebook-native-login#2]
Thanks for this, it works perfectly for me on Android!

However, when I run in xcode, I get the following error when I call loginWithNativeFacebook:

`Uncaught Error: Error: Can't call 'facebookConnectPlugin.login'. You aren't under Cordova environment.:111:http://localhost:12744/packages/jsep_facebook-native-login.js?hash=121ea903de803856be4db363d87a66ca286a5ec8`

Now, I'm new to IOS, Cordova, and xcode, but my logs indicate that I am indeed in a cordova environment with the following:

`Apache Cordova native platform version 4.1.0 is starting.`

**Comment by rohanray** [comment]
@jsep I have the same issue as mentioned by @DavidFishman . It's working fine on Android but not in ios. I am running meteor run ios-device, and selecting my own iphone and not a simulator to run the project, but still land up with the same issue `Uncaught Error: Error: Can't call 'facebookConnectPlugin.login'. You aren't under Cordova environment.`

**Comment by jsep** [comment]
Hello @DavidFishman, @rohanray 

I don't have a Mac I can't test this, but I throw the error I can't find the `facebookConnectPlugin` as global object.

**Comment by Kostanos** [comment]
same error here. so `global.facebookConnectPlugin` is not defined for some reason?

**Comment by dohomi** [comment]
Same error happens here. Any idea how to solve it? Thanks

**Comment by GManzato** [comment]
in facebookConnectPlugin.js , the object global.facebookConnectPlugin is not yet available when the code run. Need to do some kind of promise/ready lookup. I've find an easy turn around , simply remove facebookConnect.js an access directly to the global object.  Look at my fork https://github.com/GManzato/meteor-facebook-native-login

**Comment by pixelsage** [comment]
@GManzato Thanks for the fork! I got it working on device, but for some reason after authenticating on simulator, it doesn't actually log me in. Do you happen to have a similar problem?

**Comment by GManzato** [comment]
@pixelsage yes, same for me. I think it's because you don't have the native app on the simulator. but work perfectly on the device, I have an app on the app store using it without any problem

**Comment by pixelsage** [comment]
@jsep You might consider pulling @GManzato 's fork into this so people who use your package will have a working version for iOS. Maybe even put a caveat in the readme so people know that they have to test it on device to see it working, as mentioned in the comment above.

**Comment by pixelsage** [comment]
@GManzato Do you plan to submit your package to atmosphere.js? If not, mind if I do so? Will make it easier for people to install it in the future, since it looks like @jsep isn't currently keeping tabs on this project.

**Comment by jsep** [comment]
Hello everyone, I'm going to release a new version very soon.

**Comment by mxab** [comment]
@jsep any update on the new version?

**Comment by Yonben** [comment]
Looking for update too..

**Comment by aedensixty** [comment]
Waiting on this fix. Using other package for now.

**Comment by Almaju** [comment]
I tried with GManzato but I got my app bundle refused because it is not native. :/

Looking for updates! Thanks for your work :)
